File: frontend/src/lib/types/scraper.ts
// TypeScript types for ScrapeResult (matches backend contract)
export interface Link {
  text: string;
  href: string;
}

export interface ScrapeResult {
  title: string;
  markdown: string;
  links: Link[];
  rawHtml?: string;
  warnings?: string[];
  fetchedAt: string;
}

// Types for the remote function return values
export interface ScrapeSuccessResult {
  url: string;
  depth: number;
  success: true;
  final: ScrapeResult;
}

export interface ScrapeErrorResult {
  url: string;
  depth: number;
  success: false;
  error: string;
}

export type ScrapeFormResult = ScrapeSuccessResult | ScrapeErrorResult;

File: backend/src/models/scraper.go
package models

import "time"

// Link represents a hyperlink discovered during scraping
type Link struct {
	Href string `json:"href"`           // Absolute URL
	Text string `json:"text,omitempty"` // Link text or anchor
}

// ScrapeResult represents the output from a scrape job
type ScrapeResult struct {
	Title     string    `json:"title"`              // Page title
	RawHTML   string    `json:"rawHtml"`            // Raw HTML content of the page
	Markdown  string    `json:"markdown"`           // Main content converted to Markdown
	Links     []Link    `json:"links"`              // Discovered links
	Warnings  []string  `json:"warnings,omitempty"` // Optional warnings (robots.txt, fallback, etc.)
	FetchedAt time.Time `json:"fetchedAt"`          // ISO-8601 timestamp
}
